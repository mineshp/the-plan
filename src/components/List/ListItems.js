import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';
import ListItemRow from './ListItemRow';

const ListItems = ({ items, handleChange }) => {
    const listItemRows = [];
    items.map((listRowObj) => (
        listItemRows.push(
            <ListItemRow
                key={listRowObj.rowId}
                itemRow={listRowObj}
                handleChange={handleChange}
            />
        )
    ));

    return (
        <Table.Body>
            {listItemRows}
        </Table.Body>
    );
};

ListItems.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({})
    ).isRequired,
    handleChange: PropTypes.func.isRequired
};

export default ListItems;
