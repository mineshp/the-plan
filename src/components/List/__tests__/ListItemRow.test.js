import React from 'react';
import renderer from 'react-test-renderer';
import ListItems from '../ListItems';
import { mockItemsData, mockCompletedItemsData } from '../../../helpers/test/testData/listData';

const mockhandleChange = jest.fn();
const mockHandleDelete = jest.fn();
const mockHandleCompleted = jest.fn();
const mockItems = mockItemsData();
const mockCompletedItems = mockCompletedItemsData();

describe('ListItems', () => {
    it('renders list items correctly', () => {
        const tree = renderer.create(
            <ListItems
                items={mockItems}
                handleChange={mockhandleChange}
                handleDelete={mockHandleDelete}
                handleCompleted={mockHandleCompleted}
            />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders list items correctly that have some completed items', () => {
        const tree = renderer.create(
            <ListItems
                items={mockCompletedItems}
                handleChange={mockhandleChange}
                handleDelete={mockHandleDelete}
                handleCompleted={mockHandleCompleted}
            />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
