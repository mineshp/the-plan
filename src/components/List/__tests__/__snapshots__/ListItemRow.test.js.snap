// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItems renders list items correctly 1`] = `
<tbody
  className=""
>
  <tr
    className=""
  >
    <td
      className="six wide"
    >
      <div
        className="ui fluid input"
      >
        <input
          disabled={undefined}
          id="001-002"
          name="Name"
          onChange={[Function]}
          tabIndex={undefined}
          type="text"
          value="Captain America"
        />
      </div>
    </td>
    <td
      className="six wide"
    >
      <div
        className="ui fluid input"
      >
        <input
          disabled={undefined}
          id="001-002"
          name="Desc"
          onChange={[Function]}
          tabIndex={undefined}
          type="text"
          value="The first avenger"
        />
      </div>
    </td>
    <td
      className="right aligned six wide"
    >
      <button
        className="ui pink icon button"
        disabled={undefined}
        id="001-002"
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="trash icon"
        />
      </button>
    </td>
  </tr>
</tbody>
`;
